File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int sumNumbers(TreeNode root) {
        return helper(root,0);
    }
    private int helper(TreeNode root,int currSum){
        if(root == null){
            return 0;
        }
        currSum = currSum * 10 + root.val;
        
        if(root.left == null && root.right == null){
            return currSum;
        }

        int leftSum = helper(root.left,currSum);
        int rightSum = helper(root.right,currSum);

        return leftSum+rightSum;
        
    }
}

File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def sumNumbers(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        def helper(root, currSum):
            if root is None:
                return 0

            currSum = currSum * 10 + root.val

            if root.left is None and root.right is None:
                return currSum  

            leftSum = helper(root.left, currSum)
            rightSum = helper(root.right, currSum)

            return leftSum + rightSum 

        return helper(root, 0)

        